src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java

```diff
@@ -235,7 +238,9 @@ public Scheduler createIoScheduler(SchedulerConfig config, int workers, Supplier
     }
     final String schedulerName = resolveIoSchedulerName(config);
     Scheduler scheduler;
-    if (config.getMaxConcurrentTasks() != null) {
+    // Only throttle if max concurrent tasks is less than total backing pool size
+    if (config.getMaxConcurrentTasks() != null
```

Comment:
extranc this repeated condition to a `shouldThottle` method, and put the comment in the previous line as its javadoc